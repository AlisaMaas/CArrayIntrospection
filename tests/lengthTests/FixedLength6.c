int m_strlen(char *a) {
    int i = 0;
    while (a[i++]);
    return i;
}
int foo (char *a, int len) {
    if (len < 0) {
        len = m_strlen(a);
    }
    int sum = 0;
    for (int i = 0; i < len; i++) {
        sum += (int)a[i];
    }
    return sum;
}