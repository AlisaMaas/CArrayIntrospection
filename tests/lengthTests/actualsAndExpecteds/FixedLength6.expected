m_strlen with argument 0 should be annotated NULL_TERMINATED (Sentinel-terminated by 0)  because found a non optional sentinel check for a in m_strlen
foo with argument 0 should be annotated Parameter length of 1.
